### ISSUE TITLE
Issue with dispatcher.on('end') with client.on('voiceStateUpdate')

### ISSUE THREAD

**Issue by Andreybest** [discordjs/discord.js#2359]
```js
const Discord = require('discord.js');
const client = new Discord.Client();

client.login('token');

client.on('voiceStateUpdate', (oldMember, newMember) => {
    if(oldMember.voiceChannel === undefined && newMember.voiceChannel != undefined) {
        newMember.voiceChannel.join()
        .then(connection => {
            const dispatcher = connection.playArbitraryInput('https://translate.google.com/translate_tts?ie=UTF-8&q=Hi!&tl=en&total=1&idx=0&textlen=3&tk=609435.998434&client=t&prev=input&ttsspeed=1');
            dispatcher.on('end', () => {
                connection.disconnect();
            });
        })
    .catch(console.log);
    }
});
```
I have a bot that connect to the same voice channel and that he plays generated tts, and than he leaves channel when his tts audio ends.
But after connecting to channel he instatly leaves and crashe with this error
```js
 Error: WebSocket was closed before the connection was established
```
As i understood it bug only with voiceStateUpdate. But if I call him with a message command it works perfectly.

**Further details:**

- discord.js version: 11.3.0
- node.js version: 8.4.0
- Operating system: Ubuntu

- [ ] I found this issue while running code on a __user account__
- [ ] I have also tested the issue on latest master, commit hash:

**Comment by ZelAk312** [comment]
I think I found the issue for this, I just had the same issue and I was wondering why it was doing this. I found out that it's because you are not ignoring the but when he join the channel so you will get two voiceStateUpdate events. If you ignore when it's your bot that is joining it should be working fine. I tested it for the stable and the master version and I never got this problem again.

**Comment by appellation** [comment]
I believe @ZelAk312 is correct, that this is caused by the bot rapidly connecting to the voice channel twice.  In this situation, such behaviour should be expected and doesn't result from incorrect handling within the library.

**Comment by hydrabolt** [comment]
@appellation is correct, if you leave whilst still connecting an error will be thrown. However, the commit above should hopefully prevent this if you leave and reconnect almost instantly.

**Comment by Raidr1212** [comment]
Experiencing the same problem on the latest version with this sample code
`async function plays(channel, soundfile) {
    const connection = await channel.join().catch();
    const random = Math.floor(Math.random() * soundfile.length);
    const dispatcher = connection.play("sounds/" + soundfile[random]);
    dispatcher.setVolume(0.5)
    dispatcher.on('finish', () => {
        connection.disconnect();
    });
}`

**Comment by kyranet** [comment]
We're working on a new implementation of Discord's Voice API that has better playback quality and is more reliable than what we currently support in Discord.js v12 - check it out at https://github.com/discordjs/voice!

The new library solves many of the issues most of the users are facing, and easy support for it has been implemented since #5402. You can also implement the adaptor yourself, or switch to `@dev` for it. If you do, please check the `#djs-new-voice` for anything related to the `@discordjs/voice` library, and `#djs-at-dev` for anything related to `discord.js@13.0.0-dev` from the [official Discord server](https://discord.gg/djs).
